### ISSUE TITLE
Failure in test_categorize_nan

### ISSUE THREAD

**Issue by jakirkham** [dask/dask#2373]
```python
_____________________________ test_categorize_nan ______________________________
[gw2] linux -- Python 3.5.3 /home/travis/miniconda/envs/test-environment/bin/python
def test_categorize_nan():
        df = dd.from_pandas(pd.DataFrame({"A": ['a', 'b', 'a', float('nan')]}),
                            npartitions=2)
        with warnings.catch_warnings(record=True) as record:
            df.categorize().compute()
>       assert len(record) == 0
E       assert 1 == 0
E        +  where 1 = len([<warnings.WarningMessage object at 0x7f2e40707160>])
dask/dataframe/tests/test_categorical.py:239: AssertionError
```

ref: https://travis-ci.org/dask/dask/jobs/235413734#L6744

**Comment by jakirkham** [comment]
This is also showing up in `master` when it wasn't showing up previously.

ref: https://travis-ci.org/dask/dask/jobs/235211930#L6736

**Comment by jcrist** [comment]
This is due to the recent pytest release. See #2370 and #2369.

**Comment by jcrist** [comment]
Fixed by #2370.
